azkaban-common/src/main/java/azkaban/project/JdbcDependencyManager.java

```diff
@@ -56,35 +58,59 @@
     // after the query completes.
     Map<String, Dependency> hashAndFileNameToDep = new HashMap<>();
-
-    PreparedStatement stmnt = this.dbOperator.getDataSource().getConnection().prepareStatement(
-        String.format("SELECT file_name, file_sha1, validation_status FROM validated_dependencies "
-            + "WHERE validation_key = ? AND (%s)", makeStrWithQuestionMarks(deps.size())));
-
-    // Set the first param, which is the validation_key
-    stmnt.setString(1, validationKey);
-
-    // Start at 2 because the first parameter is at index 1, and that is the validator key that we already set.
-    int index = 2;
-    for (Dependency d : deps) {
-      stmnt.setString(index++, d.getFileName());
-      stmnt.setString(index++, d.getSHA1());
-      hashAndFileNameToDep.put(d.getFileName() + d.getSHA1(), d);
+    Connection conn = null;
+    ResultSet rs = null;
+    PreparedStatement stmnt = null;
+
+    // TODO: Use azkaban.db.DatabaseOperator.query() instead of getting the DB connection and
+    // dealing with connection lifecycle.
+
+    try {
+      conn = this.dbOperator.getDataSource().getConnection();
+      if (conn == null) {
+        throw new SQLException("Null connection");
+      }
+      conn.setAutoCommit(false);
+      stmnt = conn.prepareStatement(
+          String
+              .format("SELECT file_name, file_sha1, validation_status FROM validated_dependencies "
+                  + "WHERE validation_key = ? AND (%s)", makeStrWithQuestionMarks(deps.size())));
+
+      // Set the first param, which is the validation_key
+      stmnt.setString(1, validationKey);
+
+      // Start at 2 because the first parameter is at index 1, and that is the validator key that we already set.
+      int index = 2;
+      for (Dependency d : deps) {
+        stmnt.setString(index++, d.getFileName());
+        stmnt.setString(index++, d.getSHA1());
+        hashAndFileNameToDep.put(d.getFileName() + d.getSHA1(), d);
+      }
+
+      rs = stmnt.executeQuery();
+      conn.commit();
+
+      while (rs.next()) {
+        // Columns are (starting at index 1): file_name, file_sha1, validation_status
+        Dependency d = hashAndFileNameToDep.remove(rs.getString(1) + rs.getString(2));
+        FileValidationStatus v = FileValidationStatus.valueOf(rs.getInt(3));
+        depValidationStatuses.put(d, v);
+      }
+
+      // All remaining dependencies in the hashToDep map should be marked as being NEW (because they weren't
+      // associated with any DB entry)
+      hashAndFileNameToDep.values().stream()
+          .forEach(d -> depValidationStatuses.put(d, FileValidationStatus.NEW));
+    } catch (final SQLException ex) {
+      log.error("Transaction failed: ", ex);
+      throw ex;
+    } finally {
+        // Replicate the order of closing in org.apache.commons.dbutils.QueryRunner#query
+        DbUtils.closeQuietly(rs);
+        DbUtils.closeQuietly(stmnt);
+        DbUtils.closeQuietly(conn);
```

Comment:
Nit 1 -  consider replacing with: `public static void closeQuietly(Connection conn, Statement stmt, ResultSet rs)` in DBUtils.
Nit 2 - include a log.debug() indicating that db connection resources are being cleaned up.

Reply:
Addressed 1. 

For 2, it will be better to add this in query(). Since this change will be replaced by query() soon, not adding it here.